cluster/node-pools/master-ubuntu-default/userdata.yaml

```diff
@@ -0,0 +1,571 @@
+          - --horizontal-pod-autoscaler-tolerance={{ .Cluster.ConfigItems.horizontal_pod_autoscaler_tolerance }}
+          - --horizontal-pod-autoscaler-upscale-delay={{ .Cluster.ConfigItems.horizontal_pod_autoscaler_upscale_delay }}
+          resources:
+            requests:
+              cpu: 100m
+              memory: 100Mi
+          livenessProbe:
+            httpGet:
+              host: 127.0.0.1
+              path: /healthz
+              port: 10252
+            initialDelaySeconds: 15
+            timeoutSeconds: 15
+          volumeMounts:
+          - mountPath: /etc/kubernetes/ssl
+            name: ssl-certs-kubernetes
+            readOnly: true
+          - mountPath: /etc/ssl/certs
+            name: ssl-certs-host
+            readOnly: true
+          - mountPath: /etc/kubernetes
+            name: kubeconfig
+            readOnly: true
+        hostNetwork: true
+        volumes:
+        - hostPath:
+            path: /etc/kubernetes/ssl
+          name: ssl-certs-kubernetes
+        - hostPath:
+            path: /etc/kubernetes
+          name: kubeconfig
+        - hostPath:
+            path: /usr/share/ca-certificates
+          name: ssl-certs-host
+
+  - owner: root:root
+    path: /etc/kubernetes/manifests/kube-scheduler.yaml
+    content: |
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: kube-scheduler
+        namespace: kube-system
+        labels:
+          application: kube-scheduler
+          version: v1.13.5-1
+      spec:
+        priorityClassName: system-node-critical
+        tolerations:
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
+        hostNetwork: true
+        containers:
+        - name: kube-scheduler
+          image: registry.opensource.zalan.do/teapot/kube-scheduler:v1.13.5-1
+          args:
+          - --master=http://127.0.0.1:8080
+          - --leader-elect=true
+          - --feature-gates=TaintBasedEvictions=true,TaintNodesByCondition={{.Cluster.ConfigItems.experimental_schedule_daemonset_pods}},ScheduleDaemonSetPods={{.Cluster.ConfigItems.experimental_schedule_daemonset_pods}}
+          env:
+          - name: KUBE_MAX_PD_VOLS
+            value: "26"
+          resources:
+            requests:
+              cpu: 100m
+              memory: 100Mi
+          livenessProbe:
+            httpGet:
+              host: 127.0.0.1
+              path: /healthz
+              port: 10251
+            initialDelaySeconds: 15
+            timeoutSeconds: 15
+
+  - owner: root:root
+    path: /etc/kubernetes/config/tokenfile.csv
+    content: |
+      {{ .Cluster.ConfigItems.worker_shared_secret }},kubelet,kubelet
+
+  - owner: root:root
+    path: /etc/kubernetes/config/image-policy-webhook.yaml
+    content: |
+      imagePolicy:
+        kubeConfigFile: /etc/kubernetes/config/image-policy-webhook-kubeconfig.yaml
+        allowTTL: 2
+        denyTTL: 2
+        retryBackoff: 500
+        defaultAllow: false
```

Comment:
Can/should this be moved to the image?

Reply:
I would really like to keep the config that we are most likely to touch to be in this repository otherwise I'm afraid we create a weird split of configuration between the AMI repository and kubernetes-on-aws which can cause confusion and make changes harder to perform (changes in two places instead of one).

For me a logical split would be to say, systemd units have to be on the AMI to achieve the desired booting sequence, but anything else config related that is not generic between master and worker should be in "userdata" in `kubenertes-on-aws`. WDYT?

Reply:
I agree. By baking our own AMI we opened a can of worms in that regard. We should define and agree upon what kind of config goes into the AMI vs. Userdata.